### PR TITLE
datalog html: Document, fix, update, and create new test data

### PR DESCRIPTION
**Pull request by microbit-carlos**
As a continuation for the changes implemented in PR https://github.com/lancaster-university/codal-microbit-v2/pull/196, this one updates the existing test files to contain more realistic raw data.

I believe the content of the `sample-*.txt` files might have been extracted from a browser save of the MY_DATA.HTM file, rather than from the straight raw data from the micro:bit. This is mostly indicated by the "empty space" being marked as `0xEFBFBD` instead of `0xFF`. In some cases, like the original `sample-trailer.txt` I could not manually fix the data, so had to create it from scratch by creating a MakeCode programme and extracting the raw data.
This is also useful because since the addition of the nRF52 Interface compatibility to CODAL some of the memory mapping has changed slightly.

I've also created additional test files and added descriptions to all of them in the README.

### CODE REVIEW

**Comment by microbit-carlos** [comment]
@microbit-matt-hillsdon Would you be able to review the PR? No need to look into the binary data, but it would be good to have a second pair of eyes to see if the README additions make sense.

File: resources/logfs/README.md [hunk 1]
```diff
@@ -23,3 +23,96 @@ An equivalently named HTML file is created for each `sample-*.txt` file in this
...
+- **sample-invalid-time-series.txt**: An invalid time series table with 4 columns (Goat, x,
+  y , z) and 172 data rows with increasing time stamps.
+    - For a time series the first column should be named `Time...`, this test
+      file was based on `sample-time-series` and the `Time (seconds)` column
+      name has been renamed to `Goat (seconds)`.
+    - In this test file the "Visual Preview" should not work.
+- **sample-time-series-nonsequential**: Based on the `sample-time-series` file,
+  a single entry has been modified to make the time stamps non-sequential.
+    - Data entry 169 (6th from the bottom) has been changed from `18.99` to
+      `99.99`. As the next entry time stamp is `19.1`, this represent a second
```
Comment (microbit-matt-hillsdon) [inline resources/logfs/README.md:97]:
```suggestion
      `99.99`. As the next entry time stamp is `19.1`, this represents a second
```

File: resources/logfs/README.md [hunk 2]
```diff
@@ -23,3 +23,96 @@ An equivalently named HTML file is created for each `sample-*.txt` file in this
...
+      set of logged data after a micro:bit reset.
+    - In this test file the "Visual Preview" should plot a graph until entry
+      169, and display some kind of warning/error message to indicate it is
+      not plotting the entire dataset.
+- **sample-full-log**: A full log with 6064 rows.
```
Comment (microbit-matt-hillsdon) [inline resources/logfs/README.md:102]: Maybe a comment to say there should be an indicator showing the log is full?

**Review by microbit-matt-hillsdon — APPROVED** [review]
Sample files, test data and readme look good to me.
A couple of trivial comments above.

File: resources/logfs/README.md (hunk 2 above)
Comment (microbit-carlos) [reply resources/logfs/README.md:102]: Yep, good idea! 👍

**Comment by microbit-carlos** [comment]
I'll squash the last three commits for a cleaner git history, as it'd be good to keep the first two separated.

**Pull request merged**